.github/workflows/docs-preview.yaml

```diff
@@ -0,0 +1,76 @@
+name: Docs Preview on PR
+
+on:
+  pull_request:
+    paths:
+      - "apps/docs/**"
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: pages
+  cancel-in-progress: false
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install PNPM
+        uses: pnpm/action-setup@v4
+        with:
+          version: 9.0.5
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: pnpm
+
+      - name: Setup Pages
+        uses: actions/configure-pages@v4
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Build project
+        run: pnpm build
+
+      - name: Build with VitePress
+        working-directory: apps/docs
+        run: pnpm build
```

Comment:
The previous build will also have built `docs`.

Reply:
Sincerely, I didn't understand your point, but I think you're talking about the two builds that the CI does.